RFC-008-release-versioning.md

```diff
@@ -0,0 +1,116 @@
+# RFC008 - Release Versioning Strategy
+
+## Proposed by
+
+Aaron Peters
+
+
+## Current status
+
+See the status indicated on the pull request description.
+
+
+## Proposal overview
+
+As The Good Docs Project moves towards a formal release structure, this RFC proposes a standard versioning scheme to apply at least to the project's Product Outputs, as defined by outputs directly employed by users in their own documentation efforts. For example, this includes the collection of templates (which users copy as a starting point for their own documents), but not whitepapers (which serve to inform their audience, but are not used verbatim).
+
+TGDP will use the Semantic Versioning convention for the versioning of individual Product Outputs (at the time of writing, this includes the template collection and the DocTools website package).
+
+> :triangular_flag_on_post: For the purposes of this RFC, "features" and "functionality" can refer to new **content**, such as additional sections or boilerplate text, added to doc templates.
+
+
+## Motivation
+
+Using the well-established [https://semver.org/](Semantic Versioning) framework will provide the following benefits:
+
+- It will demonstrate the project's progress to users and prospective users through publicly visible notifications.
+- It will set expectations to current users regarding the extent of change(s) contained in a newly-release version of a Product Output, as compared to the version the user currently has, including potentially breaking backward-compatibility.
+- It provides contributors to the project a common point of reference when collaborating, making the team more efficient.
+- It provides downstream integrators of TGDP's Product Outputs an easy way to identify if their own work products will need to be updated to support new features, or if there is a risk of compatibility issues.
+
+
+## Proposal

```

Comment:
For the content that is in this proposal section, I think it is a good foundation, and is pretty close to right. I also think we should extend it to cover:
* From a general documentation point of view (say for docs which describe a product), what constitutes a sematic update based on major.minor.patch. What is:
  * A 1.0 release
  * A major update
  * A minor update
  * A patch update
* Same questions, but this time for our template.
* I think we should also discuss the difference between a document update, and a documentation set update. (Analogy is a package version vs a linux distribution version.)

Reply:
Okay, I've [drafted a proposal for how semantic versioning should apply to docs](https://docs.google.com/document/d/1gPr_OHoUkg5Wjj3UWYX78ORtTBArSJck4mWrKhWpfPA/edit#).

Reply:
I made a mistake and merged this in, but I've noted your comments in the "Feedback" section of the new RFC. I'll let Aaron decide if he wants to move it up into the body of the RFC.